helios-client/src/test/java/com/spotify/helios/common/JobValidatorTest.java

```diff
@@ -596,4 +601,30 @@ public void testJobFromJsonWithInvalidRegistration() throws Exception {
         contains("registration for 'volumes' is malformed: does not have a port mapping"));
   }
 
+  /**
+   * Tests that Jobs cannot have both grace period and external ports.
+   */
+  @Test
+  public void testValidateGracePeriodAndExternalPorts() throws Exception {
+    final Job noPorts = Json.read(stringFromResource("job-with-no-ports.json"), Job.class);
+    assertThat(validator.validate(noPorts), Matchers.<String>empty());
+
+    final Job noExternalPorts = Json.read(
+        stringFromResource("job-with-no-external-ports.json"), Job.class);
+    assertThat(validator.validate(noExternalPorts), Matchers.<String>empty());
+
+    final Job externalPorts = Json.read(
+        stringFromResource("job-with-external-ports.json"), Job.class);
+    assertThat(validator.validate(externalPorts), Matchers.<String>empty());
+
+    final Job externalPortsAndGracePeriod = Json.read(
+        stringFromResource("job-with-external-ports-and-grace-period.json"), Job.class);
+    assertThat(validator.validate(externalPortsAndGracePeriod),
+        hasItem("Job has a grace period and static/external ports. "
+                + "Please choose one or the other."));
```

Comment:
is there a test for grace period and no static ports?

Reply:
Thanks. Added.